src/map/lua/lua_baseentity.cpp

```diff
@@ -16660,6 +16679,40 @@ bool CLuaBaseEntity::hasTPMoves()
     return !MobSkills.empty();
 }
 
+/************************************************************************
+ *  Function: drawIn()
+ *  Purpose : Draws in the target, or current target if not specified
+ *  Example : mob:drawIn()     mob:drawIn(player)
+ *  Notes   : Draws in a player even if within the draw-in leash
+ ************************************************************************/
+void CLuaBaseEntity::drawIn(sol::variadic_args va)
+{
+    auto mobObj = dynamic_cast<CMobEntity*>(m_PBaseEntity);
+
+    if (va.size() == 0)
+    {
+        auto defaultTarget = mobObj->GetBattleTarget();
+
+        if (defaultTarget == nullptr)
+        {
+            return;
+        }
+        battleutils::DrawIn(defaultTarget, mobObj, mobObj->GetMeleeRange() - 0.2f);
+        return;
+    }
+
+    CBattleEntity*  PTarget{ nullptr };
+    CLuaBaseEntity* PLuaBaseEntity = va.get<CLuaBaseEntity*>(0);
```

Comment:
Presumably this will return a nullptr if you pass in a number to the function, right?

Reply:
The following line should fix this no?
```cpp
PTarget                        = PLuaBaseEntity ? (CBattleEntity*)PLuaBaseEntity->m_PBaseEntity : nullptr;
```
If `PLuaBaseEntity != nullptr`, then `PTarget` is assigned the BaseEntity
Otherwise if `PLuaBaseEntity` was `nullptr`, then `PTarget` is assigned `nullptr`.

Then immediately after that:
```cpp
if (PTarget)   // If PTarget is NULL, we skip this segment
{ [ ..snip..] }
return;        // and end up returning
```

Or are you saying that `va.get<CLuaBaseEntity*>(0)` will crash on the invalid argument? (IE: int instead of Player/Target object)

Reply:
avoid usage of C style casts, 

`(CBattleEntity*)PLuaBaseEntity->m_PBaseEntity` -> `dynamic_cast<CBattleEntity*>(PLuaBaseEntity->m_PBaseEntity)`

Reply:
CLuaBaseEntity is not polymorphic 

Implementing that suggestion generates the following error:
`the operand of a runtime dynamic_cast must have a polymorphic class type`

Reply:
try `dynamic_cast<CBattleEntity*>(PLuaBaseEntity->GetBaseEntity())`

Reply:
Re-Addressed both of these with making it a bit more explicit in the source
```cpp
    CLuaBaseEntity* PLuaBaseEntity = va.get<CLuaBaseEntity*>(0);

    if (!PLuaBaseEntity)
    {
        ShowError("Attempt to draw-in non-valid target.");
        return;
    }

    CBaseEntity* PBaseEntity = PLuaBaseEntity->m_PBaseEntity;
    CBattleEntity* PTarget = nullptr;
    
    if (PBaseEntity)
    {
        PTarget = dynamic_cast<CBattleEntity*>(PBaseEntity);
    }

    if (PTarget)
    {
        battleutils::DrawIn(PTarget, mobObj, mobObj->GetMeleeRange() - 0.2f);
    }

    return;
}
 ```
 
 RE: Zach's initial question -- If a Number is passed into this function, a LUA error in generated in the xi_map.exe process.  No crashes occur in the actual map process though.
![image](https://github.com/LandSandBoat/server/assets/47504230/2a0190b5-85c9-4ea1-aeac-07f5e58353d0)